scripts/organismSpecies/parse_species.py

```diff
@@ -0,0 +1,203 @@
+                    short_name='f')
+
+flags.DEFINE_string('old_mcf',
+                    'Species.mcf',
+                    'Old species file path.',
+                    short_name='p')
+
+flags.DEFINE_string('output_mcf',
+                    'newSpecies.mcf',
+                    'The output data mcf file path.',
+                    short_name='m')
+
+
+def get_mcf_piece(code, info, seen_name_to_dcid, CODE_TO_KINGDOM):
+    """
+    Args:
+        code: an organism code
+        info: an information map
+    Returns:
+        a data mcf text.
+    Example: code = 'AADNV',
+    info = {'Official': 'Abaeis nicippe', 'kingdom': 'E',
+    'taxonID': '72259', 'Common': 'Sleepy orange butterfly'}
+    """
+    name = get_class_name(info['Official'])
+
+    name_seen = False
+    # if this species is already in Species.mcf in DCs, keep dcid
+    if name in seen_name_to_dcid:
+        dcid = seen_name_to_dcid[name]
+        name_seen = True
+    else:
+        dcid = code
+    type_names = ''
+    if 'Official' in info:
+        type_names += 'scientificName: "' + info['Official'] + '"\n'
+    if 'Common' in info:
+        type_names += 'commonName: "' + info['Common'] + '"\n'
+    if 'Alternate' in info:
+        type_names += 'alternateName: "' + info['Alternate'] + '"\n'
+    taxonID = info['taxonID']
+    if taxonID == "31609":
+        # scientific name = "Parainfluenza virus 5 (isolate Canine/CPI-)"
+        name += 'Negative'
+    elif taxonID == "31608":
+        # scientific name = "Parainfluenza virus 5 (isolate Canine/CPI+)"
+        name += 'Positive'
+    mcf = ('Node: ' + name + '\n' + 'name: "' + name + '"\n' +
+           'typeOf: Species' + '\n' + type_names + 'ncbiTaxonID: "' + taxonID +
+           '"\n' + 'organismTaxonomicKingdom: dcs:' +
+           CODE_TO_KINGDOM[info['kingdom']] + '\n' + 'uniProtOrganismCode: "' +
+           code + '"\n' + 'dcid: "bio/' + dcid + '"\n')
+
+    return mcf, name_seen
+
+
+def get_class_name(a_string):
+    """Convert a name string to format: ThisIsAnUnusualName.
+    Take a space delimited string, return a class name such as ThisIsAnUnusualName
+    Here we use this function for instance name. Thus it allows to start with a number
+    """
+    joint_name = a_string.title().replace(' ', '')
+    # substitute except for  _, character, number
+    non_legitimate = re.compile(r'[\W]+')
+    class_name = non_legitimate.sub('', joint_name)
+    return class_name
+
+
+def get_mcf(combine_lines, seen_name_to_dcid):
+    """Generate mcf list.
+    Args:
+        combine_lines: a list, each contains a line from the database
+        seen_name_to_dcid: a dict mapping seen species name to dcid
+    Returns:
+        mcf_seen_list: new schema list of the species already in DC
+        mcf_list: new schema list of the species not in DC
+    """
+
+    mcf_list = []
+    mcf_seen_list = []
+    code = None
+    name_type_map = {'N': 'Official', 'C': 'Common', 'S': 'Alternate'}
+    CODE_TO_KINGDOM = {
```

Comment:
nit: If this is only used in get_mcf_piece() why not just define it there instead of passing it as arg?

Reply:
Great catch!